src/vsphere_cpi/lib/cloud/vsphere/vcenter_client.rb

```diff
@@ -210,6 +211,21 @@ def find_all_stemcell_replicas_in_datastore(datacenter_mob, stemcell_id, datasto
       matches
     end
 
+    def find_network_retryably(datacenter, network_name)
+      begin
+        Bosh::Retryable.new(
+            tries: 62, #total retry time - 10 minutes
+            on: [NetworkNotFoundError]
+        ).retryer do |i|
+          logger.info("Trying to find network #{i}th time")
+          network_mob = find_network(datacenter, network_name)
+          raise NetworkNotFoundError if network_mob.nil
+        end
+      rescue Bosh::Common::RetryCountExceeded => e
+        raise "Error: #{e} in finding network '#{network_name}' after multiple retries. Verify that the portgroup exists."
```

Comment:
Can we just raise an error form a string?

Reply:
This will raise a runtime error and not the error we actually rescued. Fixed it.